### PR TITLE
If and Boolean nodes

### PR DESCRIPTION
**Pull request by gmalette**
ping @sgnr

### CODE REVIEW

File: lib/liquider/ast.rb [hunk 1]
```diff
@@ -67,5 +77,9 @@ def visit(visitor)
   NullNode = Node.new_type(:null) do
     def visit(visitor)
     end
+
+    def op?
```
Comment (gmalette) [inline lib/liquider/ast.rb]: This is only so we can avoid empty `else` after each `if` that don't have their own `else`

Comment (gmalette) [inline lib/liquider/ast.rb]: In theory we could make it more intelligent and avoid branches that will never `op`

Comment (gmalette) [inline lib/liquider/ast.rb]:
```ruby
ElseNode.new(
  DocumentNode.new([
    StringNode.new(""),
  ]),
)
```
could no-op

File: lib/liquider/ast.rb [hunk 2]
```diff
@@ -35,12 +39,18 @@ def visit(visitor)
           end
         VISIT
 
+        type.class_eval(&block) if block
```
Comment (sgnr) [inline lib/liquider/ast.rb]: :+1:

File: lib/liquider/ast.rb [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
   class Node
     class << self
-      def new_type(type_name, *attributes)
+      def new_type(type_name, *attributes, &block)
         type = Class.new(Node) do
           attributes.each do |attribute|
             attr_reader attribute
           end
+
+          def op?
```
Comment (sgnr) [inline lib/liquider/ast.rb]: If I understand correctly, `op? == true` means that the node will output something?

File: lib/liquider/ast.rb (hunk 1 above)
Comment (sgnr) [inline lib/liquider/ast.rb]: From the AST, having an empty else is fine... does it really uglifies the ERB?

File: lib/liquider/erb_compiler.rb [hunk 4]
```diff
@@ -133,6 +137,25 @@ def on_for(for_node)
...
+    erb_tag {
+      @output << "if "
+      if_node.head.visit(self)
+    }
+    if_node.body.visit(self)
+    if if_node.continuation.op?
+      erb_tag {
+        @output << "else"
+      }
+      if_node.continuation.visit(self)
```
Comment (sgnr) [inline lib/liquider/erb_compiler.rb]: Can we check here that this is not empty instead of `op?`.

**Comment by sgnr** [comment]
The summarize my comments, I do not like `op?` because I feel `op? == false` is redundant with an empty output, but I'm open to it if it really solves a practical problem with ERB generation.

**Comment by gmalette** [comment]
I'll remove `op?`

**Comment by gmalette** [comment]
As discussed, I removed `op?` at the cost of empty `else` branches

**Comment by sgnr** [comment]
:shipit:

**Pull request merged**